File: frontend/segmentation_fault/api/schemas/comments.ts
export interface Comment {
    id: number;
    created: Date;
    updated?: Date;
    author: number;
    content: string;
    post: number;
    votes: number;
    vote: boolean | null;
}

export interface CommentCreate {
    content: string;
    post: number;
}

export interface CommentExternal {
    id: number;
    created: string;
    updated?: string;
    author: number;
    content: string;
    post: number;
}

export interface CommentCreateExternal {
    content: string;
    post: number;
}

export interface CommentUpdate {
    content: string;
}

export interface CommentUpdateExternal {
    content: string;
}

File: backend/segmentation_fault/schemas/comments.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel


class Comment(BaseModel):
    id: int
    created: datetime
    updated: Optional[datetime]
    author: int
    content: str
    post: int


class CommentCreate(BaseModel):
    content: str
    post: int


class CommentUpdate(BaseModel):
    content: str
